eth_worker/eth_manager/blockchain_sync/blockchain_sync.py

```diff
@@ -0,0 +1,145 @@
+        'recipient_blockchain_address': transaction.recipient_address,
+        'blockchain_transaction_hash': transaction.hash,
+        'transfer_amount': float(transaction.amount)/(10**18), # Note: Tweak this later
+        'contract_address': transaction.contract_address
+    }
+    r = requests.post(config.APP_HOST + '/api/v1/credit_transfer/internal/',
+                      json=body,
+                      auth=HTTPBasicAuth(config.INTERNAL_AUTH_USERNAME,
+                                         config.INTERNAL_AUTH_PASSWORD)
+                    )
+    return r
+
+# Get list of filters from redis
+def synchronize_third_party_transactions():
+    filters = persistence_module.get_all_filters()
+    # Since the webook will timeout if we ask for too many blocks at once, we have to break 
+    # the range we want into chunks.
+    for f in filters:
+        ceiling = 0
+        max_enqueued_block = f.max_block or 0
+        latest_block = get_latest_block_number()
+        number_of_blocks_to_get = (latest_block - max_enqueued_block)
+        # integer division, then add a chunk if there's a remainder
+        number_of_chunks = int(number_of_blocks_to_get/sync_const.BLOCKS_PER_REQUEST) + (number_of_blocks_to_get % sync_const.BLOCKS_PER_REQUEST > 0)
+        for chunk in range(number_of_chunks):
+            floor = max_enqueued_block + (chunk * sync_const.BLOCKS_PER_REQUEST)
+            ceiling = max_enqueued_block + ((chunk + 1) * sync_const.BLOCKS_PER_REQUEST)
+            if ceiling > latest_block:
+                ceiling = latest_block
+            # filter_job objects are just filter objects with floors/ceilings set
+            job = {
+                'filter_id': f.id,
+                'floor': floor,
+                'ceiling': ceiling
+            }
+            red.rpush(sync_const.THIRD_PARTY_SYNC_JOBS, json.dumps(job))
+        if ceiling:
+            # Sometimes there won't be a ceiling, if there's no new chunks since the last time it ran
+            persistence_module.set_filter_max_block(f.id, ceiling)
+        # With the jobs set, we can now start processing them
+        process_all_chunks()
+
+
+# Iterates through all jobs made by synchronize_third_party_transactions
+# Gets and processes them all!
+def process_all_chunks():
+    for filter_job in red.lrange(sync_const.THIRD_PARTY_SYNC_JOBS, 0, -1):
+        task = json.loads(filter_job)
+        filter = persistence_module.get_filter(task['filter_id'])
+        transaction_history = get_blockchain_transaction_history(
+            filter.contract_address, 
+            task['floor'], 
+            task['ceiling'], 
+            filter.filter_parameters,
+            filter.id
+        )
+        for transaction in transaction_history:
+            handle_transaction(transaction)
+        # Once a batch of chunks is completed, we can mark them completed
+        persistence_module.set_block_range_status(task['floor'], task['ceiling'], 'SUCCESS')
```

Comment:
The block statuses are set in a range because a range of blocks will succeed or fail as a range. I.e. if I ask for "All transactions on contract `0xDEADBEEF` from `0 to 100`" and then I get `[]` back, then I successfully covered that `0 to 100` range.   
In other words: I don't get individual feedback for every block I request transactions from. It's either full-fail, or full-results.   
  
With all that said, I did realize it made sense to move this line to `get_blockchain_transaction_history`, since `block_status` is concerning the status of actually _getting_ the data. Synchronization status is handled by `is_synchronized_with_app`